auth/rest_http_client.hh

```diff
@@ -0,0 +1,196 @@
+        std::string _server;
+        int _port;
+        std::string _ca_file;
+        shared_ptr <tls::certificate_credentials> _creds;
+        sstring _request;
+
+    public:
+        rest_http_client() {};
+
+        rest_http_client(std::string server, int port, std::string ca_file)
+                : _server(server), _port(port), _ca_file(ca_file),
+                  _creds(make_shared<tls::certificate_credentials>()) {}
+
+        class connection {
+        private:
+            connected_socket _fd;
+            input_stream<char> _read_buf;
+            output_stream<char> _write_buf;
+            rest_response_parser _parser;
+            sstring _request;
+
+            future <std::vector<std::string>> extract_groups(temporary_buffer<char> buf, uint content_len) {
+                const char *json = buf.get();
+                picojson::value v;
+                std::string err;
+                picojson::parse(v, json, json + content_len, &err);
+                if (!err.empty()) {
+                    std::throw_with_nested(exceptions::authentication_exception(
+                            format("Bad response: not able to parse body: {}", err)));
+                }
+
+                // check if the type of the value is "object"
+                if (!v.is<picojson::object>()) {
+                    std::throw_with_nested(exceptions::authentication_exception(
+                            "Bad response: response is not a dict"));
+                }
+
+                const picojson::value::object &obj = v.get<picojson::object>();
+                auto it = obj.find("groups");
+                if (it == obj.end()) {
+                    std::throw_with_nested(exceptions::authentication_exception(
+                            "Bad response: does not contains groups field"));
+                }
+
+                auto pv_groups = it->second;
+                if (!pv_groups.is<picojson::array>()) {
+                    std::throw_with_nested(exceptions::authentication_exception(
+                            "Bad response: groups field doesn't contain an array"));
+                }
+
+                const picojson::value::array &p_groups = pv_groups.get<picojson::array>();
+                std::vector <std::string> groups;
+
+                // TODO filter groups to add (For ex. only add groups containing scylla... to avoid storing all groups)
+                for (auto p_group : p_groups) {
+                    if (p_group.is<std::string>()) {
+                        auto group = p_group.get<std::string>();
+                        groups.push_back(group);
+                    }
+                }
+
+                return make_ready_future < std::vector < std::string >> (groups);
+            }
+
+            future <std::vector<std::string>>
+            get_groups_from_body(std::unique_ptr <auth::http_response, std::default_delete<auth::http_response>> rsp) {
+                auto it = rsp->_headers.find("content-length");
```

Comment:
TODO for later: make case insensitive lookup

Reply:
Not needed the header have been fully converted to lower case in rest_response_parser.rl
```
action assign_field {
    std::transform(_field_name.begin(), _field_name.end(), _field_name.begin(), ::tolower);
...
```

Reply:
ok I missed it